lib/charms/operator_libs_linux/v1/snap.py

```diff
@@ -393,6 +393,20 @@ def connect(
         except CalledProcessError as e:
             raise SnapError("Could not {} for snap [{}]: {}".format(_cmd, self._name, e.stderr))
 
+    def hold(self, duration: Optional[str] = None) -> None:
```

Comment:
Can we add a property called `held`? To be used as 

```python
if some_snap.held:
   do_something()     
else:
  do_something_else()
```

Reply:
I like this idea. I think the signature should be to return a `bool`, rather than a hold time, for simplicity. (If later on we discover we need to know the exact `hold_time`, we can add that, but I doubt we will.

I'm not sure the best way to get that information programmatically, parsing `snap info <snapname>` is one way, I don't know if there are better ones.